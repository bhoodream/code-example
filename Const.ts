import { EmptyRatOption } from './types';

export const DEFAULT_NICK_NAME = 'Uncool Rat with no Nickname';
export const EMPTY_RAT_OPTION: EmptyRatOption = { val: '', label: 'No Rat' };
