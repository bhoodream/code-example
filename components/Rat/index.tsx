import React from 'react';

import { DEFAULT_NICK_NAME } from '../../Const';

export interface RatData {
    width: number;
    height: number;
    nickname?: string;
}

const Rat = (props: RatData) => {
    const { width, height, nickname = DEFAULT_NICK_NAME } = props;

    return (
        <div>
            <div>width: {width}</div>
            <div>height: {height}</div>
            <div>nickname: {nickname}</div>
        </div>
    );
};

export default Rat;
