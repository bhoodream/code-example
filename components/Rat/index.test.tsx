import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import Rat from './index';

import { DEFAULT_NICK_NAME } from '../../Const';

const DATA = { width: 1, height: 2, nickname: 'test' };
const { nickname, ...DATA_NO_NICKNAME } = DATA;

describe('Rat', () => {
    it('renders without crashing', () => {
        render(<Rat {...DATA} />);
    });

    it('renders default nickname', () => {
        expect(
            render(<Rat {...DATA_NO_NICKNAME} />).getByText(
                `nickname: ${DEFAULT_NICK_NAME}`
            )
        ).toBeDefined();
    });

    it('snapshots', () => {
        const tree = renderer.create(<Rat {...DATA} />).toJSON();

        const treeWithoutNickName = renderer
            .create(<Rat {...DATA_NO_NICKNAME} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
        expect(treeWithoutNickName).toMatchSnapshot();
    });
});
