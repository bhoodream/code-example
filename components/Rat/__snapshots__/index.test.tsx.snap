// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rat snapshots 1`] = `
<div>
  <div>
    width: 
    1
  </div>
  <div>
    height: 
    2
  </div>
  <div>
    nickname: 
    test
  </div>
</div>
`;

exports[`Rat snapshots 2`] = `
<div>
  <div>
    width: 
    1
  </div>
  <div>
    height: 
    2
  </div>
  <div>
    nickname: 
    Uncool Rat with no Nickname
  </div>
</div>
`;
